contract_compassion/model/contracts.py

```diff
@@ -0,0 +1,720 @@
+        # Sponsorship case
+        if contract.type == 'S':
+            next_date = datetime.strptime(contract.next_invoice_date, DF)
+            next_date += relativedelta(months=+1)
+            return next_date
+        else:
+            return super(recurring_contract, self)._compute_next_invoice_date(contract)
+
+    ################################
+    #        PUBLIC METHODS        #
+    ################################
+    def create(self, cr, uid, vals, context=None):
+        """Link child to sponsor.
+        """
+        child_id = vals.get('child_id')
+        if child_id:
+            self.pool.get('compassion.child').write(
+                cr, uid, child_id, {
+                    'sponsor_id': vals['partner_id']}, context)
+        return super(recurring_contract, self).create(cr, uid, vals, context)
+
+    def copy(self, cr, uid, id, default=None, context=None):
+        if not default:
+            default = dict()
+        num_pol_ga = self.browse(cr, uid, id, context=context).num_pol_ga
+        default.update({
+            'child_id': False,
+            'activation_date': False,
+            'is_active': False,
+            'num_pol_ga': num_pol_ga + 1,
+        })
+        return super(recurring_contract, self).copy(cr, uid, id, default,
+                                                    context)
+
+    def unlink(self, cr, uid, ids, context=None):
+        for contract in self.browse(cr, uid, ids, context):
+            # We can only delete draft contracts.
+            if contract.state != 'draft':
+                raise orm.except_orm(_('Warning'),
+                                     _('You cannot delete a contract that is '
+                                       'still active. Terminate it first.'))
+            else:
+                if contract.child_id:
+                    contract.child_id.write({'sponsor_id': False})
+
+        super(recurring_contract, self).unlink(cr, uid, ids,
+                                               context=context)
+        return
+
+    def write(self, cr, uid, ids, vals, context=None):
+        """ Perform various checks when a contract is modified. """
+        if 'child_id' in vals:
+            self._on_change_child_id(cr, uid, ids, vals['child_id'],
+                                     vals.get('partner_id'), context)
+        if 'group_id' in vals:
+            self._on_change_group_id(cr, uid, ids, vals['group_id'], context)
+
+        # Write the changes
+        res = super(recurring_contract, self).write(cr, uid, ids, vals,
+                                                    context=context)
+
+        if 'contract_line_ids' in vals:
+            self._on_contract_lines_changed(cr, uid, ids, context)
+        if 'group_id' in vals or 'partner_id' in vals:
+            self._on_group_id_changed(cr, uid, ids, context)
+
+        return res
+
+# just to modify access rights...
+class recurring_invoicer(orm.Model):

```

Comment:
I saw that these overrides are useless. You can simply find the model in the security.xml file by putting module_name.model_name.